wordpress-mobile/WordPress-Android
# Removes ImageSettingsDialogFragment which is not used anymore

## Description
This PR removes `ImageSettingsDialogFragment` which is not used anymore as identified in https://github.com/wordpress-mobile/WordPress-Android/pull/19568#pullrequestreview-1744379496

-----

## To Test:

* Verify that the app builds and the CI checks pass
* Verify that the editor opens and behaves as expected

-----

## Regression Notes

1. Potential unintended areas of impact

 Editor

2. What I did to test those areas of impact (or what existing automated tests I relied on)

 Manual testing

3. What automated tests I added (or what prevented me from doing so)

 N/A since this PR involves code cleanup

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
